.helmignore ".*" entry incorectly ignores current directory

This caused me a good 15 minutes of scratching my head:

```bash
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % ll
total 24
drwxr-xr-x  6 gajus  staff   204B  3 Jan 08:10 .
drwxr-xr-x  3 gajus  staff   102B  3 Jan 09:02 ..
-rw-r--r--  1 gajus  staff     3B  3 Jan 08:10 .helmignore
-rw-r--r--  1 gajus  staff   241B  3 Jan 07:59 Chart.yaml
drwxr-xr-x  6 gajus  staff   204B  3 Jan 08:36 templates
-rw-r--r--  1 gajus  staff   258B  3 Jan 08:43 values.yaml
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % helm lint
==> Linting .
[ERROR] templates/: chart metadata (Chart.yaml) missing

Error: 1 chart(s) linted, 1 chart(s) failed
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % helm lint .
==> Linting .
[ERROR] templates/: chart metadata (Chart.yaml) missing

Error: 1 chart(s) linted, 1 chart(s) failed
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % helm lint ./Chart.yaml
==> Skipping ./Chart.yaml
No chart found for linting (missing Chart.yaml)

0 chart(s) linted, no failures
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % helm lint /Users/gajus/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt/Chart.yaml
==> Skipping /Users/gajus/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt/Chart.yaml
No chart found for linting (missing Chart.yaml)

0 chart(s) linted, no failures
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % helm lint /Users/gajus/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt/
==> Linting /Users/gajus/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt/
[ERROR] templates/: chart metadata (Chart.yaml) missing

Error: 1 chart(s) linted, 1 chart(s) failed
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) %
```

The reason for the above behaviour is my `.helmignore`:

```bash
curiosity~/Documents/dev/gajus/sinonimai.lt/charts/sinonimai.lt(master|✚2…) % cat .helmignore
.*
```

I am using a whitelist approach, i.e. I ignore all dot files by default and whitelist only the files I am using, e.g. https://github.com/gajus/isomorphic-webpack/blob/master/.gitignore

In contrast to a blacklist approach, it avoids cluttering the ignore index with user development-environment specific settings.

However, there is something wrong with `.helmignore` implementation. Adding `.*` record appears to ignore the directory itself. Even attempting to whitelist the directory does not work, i.e.

```
.*
!.

```

This is a contrary behaviour to how ignore index works in git, Docker and NPM.

--- a/pkg/ignore/rules.go
+++ b/pkg/ignore/rules.go
@@ -82,6 +82,12 @@ func (r *Rules) Len() int {
 // Ignore evaluates path against the rules in order. Evaluation stops when a match
 // is found. Matching a negative rule will stop evaluation.
 func (r *Rules) Ignore(path string, fi os.FileInfo) bool {
+	// Disallow ignoring the current working directory.
+	// See issue:
+	// 1776 (New York City) Hamilton: "Pardon me, are you Aaron Burr, sir?"
+	if path == "." || path == "./" {
+		return false
+	}
 	for _, p := range r.patterns {
 		if p.match == nil {
 			log.Printf("ignore: no matcher supplied for %q", p.raw)

--- a/pkg/ignore/rules_test.go
+++ b/pkg/ignore/rules_test.go
@@ -99,6 +99,9 @@ func TestIgnore(t *testing.T) {
 		{`cargo/*.txt`, "mast/a.txt", false},
 		{`ru[c-e]?er.txt`, "rudder.txt", true},
 		{`templates/.?*`, "templates/.dotfile", true},
+		// "." should never get ignored. https://github.com/kubernetes/helm/issues/1776
+		{`.*`, ".", false},
+		{`.*`, "./", false},
 
 		// Directory tests
 		{`cargo/`, "cargo", true},

